### DISCUSSION TITLE
Dunst not working with DMenu

### DISCUSSION THREAD

**Discussion by lordofscripts** [dunst-project/dunst#1361] [Q&A]
- **Dunst** is installed in my Debian Raspberry π
- **DMenu** is installed
- `Dunstrc` file has `dmenu` option with correct path and `-p dunst:` as parameter.

I fire the notifications with 2 actions. _The notification pops up_ in the notification area with an `(A)` indicator; therefore it's aware of the actions. The actions have a key and a label as per dbus.

But when I middle-click on it (configured to invoke actions) I see this happen:

- all open notifications get closed instead of just the one I clicked (I posted another thread for that)
- **DMenu** is invoked by **Dunst** but only with the parameters given in the `dunstrc` configuration. It passed nothing regarding the actions it is aware of, but just sent the parameter with the prompt string.

**Comment by bynect** [comment]
Are you sure the keybindings are correctly configured?

Can you send a log ?

**Reply by lordofscripts** [reply]
I don't think it's a key-binding issue. When `dunst` invoked `dmenu` when the action was fired, `dmenu` was only called with the "`-p dunst:`" prompt parameter (shipped configuration). That is, it did **not** instruct `dmenu` which _menu items_ it should display.

**Comment by bynect** [comment]
@lordofscripts did you resolve in the end?

**Reply by lordofscripts** [reply]
Never got that to work. Updating Dunst on my Raspberry was not quite viable with my resources, couldn't compile it for my Debian ARMv7

**Reply by bynect** [reply]
I see. do action never work in general? what about links in the notification.
also, you could compile dunst with a crosscompiler for the arm. but probably the solution is something else?

**Reply by lordofscripts** [reply]
> I see. do action never work in general? what about links in the notification.
> also, you could compile dunst with a crosscompiler for the arm. but probably the solution is something else?

I get notifications but usually when not bundled (individual notifications) they get closed as if they were bundled. I think that was fixed in a later release.

Firing DMenu wasn't working either, it was configured by default in the global configuration but never actually fired dmenu.

The compile issue is because I would need to download a whole lot of dev. stuff to compile it and make config wasn't getting through. My system is small. I don't know why RASPI's Debian has such an old (not-updated) version of Dunst. Anyway, maybe one day they update that. For the time being I decided to drop the idea of notifications.

**Reply by bynect** [reply]
maybe the raspbaian uses an old debian repository? I am not a debian user but maybe you can add a custom repository/ppa with apt that uses more recent packages? or maybe you can donwload a compiled version of the package in a .deb file for your arch
